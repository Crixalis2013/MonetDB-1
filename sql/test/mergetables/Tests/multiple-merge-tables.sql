create table t (i int);
insert into t values (42), (666);
create merge table mt1 (like t);
create merge table mt2 (like t);

create temp table mycount(cc BIGINT) ON COMMIT PRESERVE ROWS;
insert into mycount SELECT (SELECT COUNT(*) FROM sys.dependencies) + (SELECT COUNT(*) FROM sys.objects);

alter table mt1 add table t;
select i from mt1;
select i from mt2; -- error, no tables associated

alter table mt2 add table t;
select i from mt1;
select i from mt2;

alter table mt1 drop table t;
select i from mt1; -- error, no tables associated
select i from mt2;

alter table mt2 drop table t;
select i from mt1; -- error, no tables associated
select i from mt2; -- error, no tables associated

SELECT CAST((SELECT COUNT(*) FROM sys.dependencies) + (SELECT COUNT(*) FROM sys.objects) - (SELECT cc FROM mycount) AS BIGINT);
	-- 0 it shouldn't have increased

drop table mycount;
drop table mt1;
drop table mt2;
drop table t;
