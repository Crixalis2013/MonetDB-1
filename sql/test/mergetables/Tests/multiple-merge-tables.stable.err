stderr of test 'multiple-merge-tables` in directory 'sql/test/mergetables` itself:


# 10:32:16 >  
# 10:32:16 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-37610" "--port=36699"
# 10:32:16 >  

MAPI  = (monetdb) /var/tmp/mtest-40339/.s.monetdb.34967
QUERY = select i from mt2; -- error, no tables associated
ERROR = !MERGE or REPLICA TABLE should have at least one table associated
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-40339/.s.monetdb.34967
QUERY = select i from mt1; -- error, no tables associated
ERROR = !MERGE or REPLICA TABLE should have at least one table associated
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-37610/.s.monetdb.36699
QUERY = select i from mt1; -- error, no tables associated
ERROR = !MERGE or REPLICA TABLE should have at least one table associated
CODE  = 42000
MAPI  = (monetdb) /var/tmp/mtest-37610/.s.monetdb.36699
QUERY = select i from mt2; -- error, no tables associated
ERROR = !MERGE or REPLICA TABLE should have at least one table associated
CODE  = 42000

# 10:32:16 >  
# 10:32:16 >  "Done."
# 10:32:16 >  

