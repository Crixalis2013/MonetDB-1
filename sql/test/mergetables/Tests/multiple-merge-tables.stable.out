stdout of test 'multiple-merge-tables` in directory 'sql/test/mergetables` itself:


# 10:32:16 >  
# 10:32:16 >  "mclient" "-lsql" "-ftest" "-tnone" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-37610" "--port=36699"
# 10:32:16 >  

#create table t (i int);
#insert into t values (42), (666);
[ 2	]
#create merge table mt1 (like t);
#create merge table mt2 (like t);
#create temp table mycount(cc BIGINT) ON COMMIT PRESERVE ROWS;
#insert into mycount SELECT (SELECT COUNT(*) FROM sys.dependencies) + (SELECT COUNT(*) FROM sys.objects);
[ 1	]
#alter table mt1 add table t;
#select i from mt1;
% sys.mt1 # table_name
% i # name
% int # type
% 3 # length
[ 42	]
[ 666	]
#alter table mt2 add table t;
#select i from mt1;
% sys.mt1 # table_name
% i # name
% int # type
% 3 # length
[ 42	]
[ 666	]
#select i from mt2;
% sys.mt2 # table_name
% i # name
% int # type
% 3 # length
[ 42	]
[ 666	]
#alter table mt1 drop table t;
#select i from mt2;
% sys.mt2 # table_name
% i # name
% int # type
% 3 # length
[ 42	]
[ 666	]
#alter table mt2 drop table t;
#SELECT CAST((SELECT COUNT(*) FROM sys.dependencies) + (SELECT COUNT(*) FROM sys.objects) - (SELECT cc FROM mycount) AS BIGINT);
% .%4 # table_name
% %4 # name
% bigint # type
% 1 # length
[ 0	]
#drop table mycount;
#drop table mt1;
#drop table mt2;
#drop table t;



# 10:32:16 >  
# 10:32:16 >  "Done."
# 10:32:16 >  

